packages/amplify-codegen-appsync-model-plugin/src/__tests__/visitors/__snapshots__/appsync-dart-visitor.test.ts.snap

```diff
@@ -0,0 +1,2131 @@
+  }
+  
+  Post.fromJson(Map<String, dynamic> json)  
+    : id = json['id'],
+      title = json['title'],
+      owner = json['owner'];
+  
+  Map<String, dynamic> toJson() =>
+    {'id': id, 'title': title, 'owner': owner};
+
+  static final QueryField ID = QueryField(fieldName: \\"id\\");
+  static final QueryField TITLE = QueryField(fieldName: \\"title\\");
+  static final QueryField OWNER = QueryField(fieldName: \\"owner\\");
+  static var schema = Model.defineSchema(define: (ModelSchemaDefinition modelSchemaDefinition) {
+    modelSchemaDefinition.name = \\"Post\\";
+    modelSchemaDefinition.pluralName = \\"Posts\\";
+    
+    modelSchemaDefinition.authRules = [
+      AuthRule(
+        authStrategy: AuthStrategy.GROUPS,
+        groupClaim: \\"cognito:groups\\",
+        groups: [ \\"admin\\" ],
+        operations: [
+          ModelOperation.CREATE,
+          ModelOperation.UPDATE,
+          ModelOperation.DELETE,
+          ModelOperation.READ
+        ]),
+      AuthRule(
+        authStrategy: AuthStrategy.OWNER,
+        ownerField: \\"owner\\",
+        identityClain: \\"cognito:username\\",
+        operations: [
+          ModelOperation.CREATE,
+          ModelOperation.UPDATE
+        ]),
+      AuthRule(
+        authStrategy: AuthStrategy.PUBLIC,
+        operations: [
+          ModelOperation.CREATE,
+          ModelOperation.UPDATE,
+          ModelOperation.DELETE,
+          ModelOperation.READ
+        ])
+    ];
+    
+    modelSchemaDefinition.addField(ModelFieldDefinition.id());
+    
+    modelSchemaDefinition.addField(ModelFieldDeinition.field(
+      key: Post.TITLE,
+      isRequired: true,
+      ofType: ModelFieldType(ModelFieldTypeEnum.string)
+    ));
+    
+    modelSchemaDefinition.addField(ModelFieldDeinition.field(
+      key: Post.OWNER,
+      isRequired: true,
+      ofType: ModelFieldType(ModelFieldTypeEnum.string)
+    ));
+  });
+}
+
+class PostType extends ModelType<Post> {
+  const PostType();
+  
+  @override
+  Post fromJson(Map<String, dynamic> jsonData) {
+    return Post.fromJson(jsonData);
+  }
+}"
+`;
+
+exports[`AppSync Dart Visitor Model with Connection Directive should generate classes for models with one to many connection 1`] = `
+"import 'package:flutter/foundation.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:collection/collection.dart';
+
+import 'Task.dart';
+
+
+/** This is an auto generated class representing the Todo type in your schema. */
+@immutalbe
+class Todo extends Model {
+  static const classType = TodoType();
+  final String id;
+  final List<Task> tasks;
+
+  @override
+  String getId() {
+    return id;
+  }
+  
+  const Todo._internal({@required this.id, this.tasks});
+  
+  factory Todo({@required String id, List<Task> tasks}) {
+    return Todo._internal(
+      id: id == null ? UUID.getUUID() : id,
+      tasks: tasks != null ? List.unmodifiable(tasks) : tasks);
```

Comment:
This should be `List.unmodifiable(tasks) : null`?

Reply:
I got this from the reference model. It will always be `null` in this case though.

Reply:
oh it's the same thing +1